org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/jarpackagerfat/FatJarRsrcUrlBuilder.java

```diff
@@ -123,16 +122,16 @@ public void writeArchive(ZipFile jarFile, IProgressMonitor progressMonitor) {
 	}
 
 	public void writeRsrcUrlClasses() throws IOException {
-		InputStream is= JavaPlugin.getDefault().getBundle().getEntry(JAR_RSRC_LOADER_ZIP).openStream();
-		ZipInputStream zis= new ZipInputStream(is);
-		ZipEntry zipEntry= zis.getNextEntry();
-		while (zipEntry != null) {
-			if (!zipEntry.isDirectory()) {
-				String entryName= zipEntry.getName();
-				byte[] content= FatJarPackagerUtil.readInputStream(zis);
-				getJarWriter().addZipEntryStream(zipEntry, new ByteArrayInputStream(content), entryName);
+		try (ZipInputStream zis= new ZipInputStream(JavaPlugin.getDefault().getBundle().getEntry(JAR_RSRC_LOADER_ZIP).openStream())){
```

Comment:
Currently eclipse.jdt.ui/org.eclipse.jdt.ui/scripts/build_jar-in-jar-loader.xml uses Java 6 level to compile & build org.eclipse.jdt.ui/jar-in-jar-loader.zip

With this change, the build fails:

```
       [javac] 3. ERROR in /data/git/eclipse.jdt.ui/org.eclipse.jdt.ui/jar in jar loader/org/eclipse/jdt/internal/jarinjarloader/JarRsrcLoader.java (at line 83)
       [javac] 	try (InputStream is =  resEnum.nextElement().openStream()){
       [javac] 	     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       [javac] Resource specification not allowed here for source level below 1.7
```

So either undo this change or address the build problem with a different PR / ticket first.

Reply:
Would it be ok to upgrade to use Java 17 for that or any reason to stick with an older version?

Reply:
AFAIK jar-in-jar loader classes is supposed to be used by **any** supported Java version users may have on classpath.
I **believe** we support running applications from IDE on Java 8 minimal (@SarikaSinha : please correct me if I'm wrong) , so that is the highest version one could use to build the jar.

Reply:
Yes @iloveeclipse

Reply:
i added an update to 1.8

Reply:
@iloveeclipse please finish review or dismiss review